crates/gitbutler-core/src/git/repository_ext.rs

```diff
@@ -79,8 +79,21 @@ impl RepositoryExt for Repository {
 
         let commit_buffer = inject_change_id(&commit_buffer, change_id)?;
 
-        let oid = do_commit_buffer(self, commit_buffer)?;
+        let should_sign = self.config()?.get_bool("commit.gpgSign").unwrap_or(false);
 
+        let oid = if should_sign {
+            match sign_buffer(self, &commit_buffer) {
+                Ok(signature) => self.commit_signed(&commit_buffer, &signature, None)?,
+                Err(e) => {
+                    tracing::warn!("Failed to sign commit, committing unsigned: {}", e);
+                    self.odb()?
+                        .write(git2::ObjectType::Commit, commit_buffer.as_bytes())?
```

Comment:
The only problem I can see immediately is that it's not going to tell the user that it's failing or why it's failing.





[3:34](https://gitbutlerworkspace.slack.com/archives/D04MS8F9ZMF/p1717421689456559)
it just won't sign things if it's misconfigured